src/main/java/seedu/address/model/student/Student.java

```diff
@@ -61,7 +61,9 @@ public boolean isSameStudent(Student otherStudent) {
 
         return otherStudent != null
                 && otherStudent.getName().equals(getName())
-                && otherStudent.getPhone().equals(getPhone());
+                && (otherStudent.getPhone().equals(getPhone())
+                    || otherStudent.getSchool().equals(getSchool())
+                        || otherStudent.getYear().equals(getYear()));
```

Comment:
Should be `&&` instead of `||` for school and year.
The student is the same only if all the field values are the same.

Reply:
okay I have resolved this. The original AB3 uses this method as a weaker notion of equality between students. Because if we check all field && then it will become the equals method. But since we defined student to have this 4 compulsory fields, we can update it to check these fields. But I think we should not include the checks for admin fields when it comes in.